File: CS/PopupMenuCustomization/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
#region #usings
using DevExpress.Utils.Menu;
using DevExpress.XtraScheduler;
using DevExpress.XtraScheduler.Services;
using DevExpress.XtraScheduler.Commands;
// ...
#endregion #usings


namespace PopupMenuCustomization
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
            Helper.FillResources(schedulerStorage1, 3);
            schedulerControl1.Views.DayView.DayCount = 2;
        }

#region #popupmenushowing
private void schedulerControl1_PopupMenuShowing(object sender, PopupMenuShowingEventArgs e)
{
    if (e.Menu.Id == DevExpress.XtraScheduler.SchedulerMenuItemId.DefaultMenu)
    {
        // Hide the "Change View To" menu item.
        SchedulerPopupMenu itemChangeViewTo = e.Menu.GetPopupMenuById(SchedulerMenuItemId.SwitchViewMenu);
        itemChangeViewTo.Visible = false;

        // Remove unnecessary items.
        e.Menu.RemoveMenuItem(SchedulerMenuItemId.NewAllDayEvent);

        // Disable the "New Recurring Appointment" menu item.
        e.Menu.DisableMenuItem(SchedulerMenuItemId.NewRecurringAppointment);

        // Find the "New Appointment" menu item and rename it.
        SchedulerMenuItem item = e.Menu.GetMenuItemById(SchedulerMenuItemId.NewAppointment);
        if (item != null) item.Caption = "&New Meeting";

        // Create a menu item for the Scheduler command.
        ISchedulerCommandFactoryService service =
        (ISchedulerCommandFactoryService)schedulerControl1.GetService(typeof(ISchedulerCommandFactoryService));
        SchedulerCommand cmd = service.CreateCommand(SchedulerCommandId.SwitchToGroupByResource);
        SchedulerMenuItemCommandWinAdapter menuItemCommandAdapter =
            new SchedulerMenuItemCommandWinAdapter(cmd);
        DXMenuItem menuItem = (DXMenuItem)menuItemCommandAdapter.CreateMenuItem(DXMenuItemPriority.Normal);
        menuItem.BeginGroup = true;
        e.Menu.Items.Add(menuItem);

        // Insert a new item into the Scheduler popup menu and handle its click event.
        e.Menu.Items.Add(new SchedulerMenuItem("Click me!", MyClickHandler));
    }
}

public void MyClickHandler(object sender, EventArgs e)
{
    MessageBox.Show("My Menu Item Clicked!");
}
#endregion #popupmenushowing
    }
}

File: VB/PopupMenuCustomization/Form1.vb
﻿Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Text
Imports System.Windows.Forms
#Region "#usings"
Imports DevExpress.Utils.Menu
Imports DevExpress.XtraScheduler
Imports DevExpress.XtraScheduler.Services
Imports DevExpress.XtraScheduler.Commands
' ...
#End Region ' #usings


Namespace PopupMenuCustomization
    Partial Public Class Form1
        Inherits DevExpress.XtraEditors.XtraForm

        Public Sub New()
            InitializeComponent()
            MySchedulerHelper.FillResources(schedulerStorage1, 3)
            schedulerControl1.Views.DayView.DayCount = 2
            schedulerControl1.Start = Date.Now.Date
            schedulerControl1.DateNavigationBar.Visible = False
            schedulerControl1.ActiveView.NavigationButtonVisibility = NavigationButtonVisibility.Never
        End Sub

#Region "#popupmenushowing"
Private Sub schedulerControl1_PopupMenuShowing(ByVal sender As Object, ByVal e As PopupMenuShowingEventArgs) Handles schedulerControl1.PopupMenuShowing
    If e.Menu.Id = DevExpress.XtraScheduler.SchedulerMenuItemId.DefaultMenu Then
        ' Hide the "Change View To" menu item.
        Dim itemChangeViewTo As SchedulerPopupMenu = e.Menu.GetPopupMenuById(SchedulerMenuItemId.SwitchViewMenu)
        itemChangeViewTo.Visible = False

        ' Remove unnecessary items.
        e.Menu.RemoveMenuItem(SchedulerMenuItemId.NewAllDayEvent)

        ' Disable the "New Recurring Appointment" menu item.
        e.Menu.DisableMenuItem(SchedulerMenuItemId.NewRecurringAppointment)

        ' Find the "New Appointment" menu item and rename it.
        Dim item As SchedulerMenuItem = e.Menu.GetMenuItemById(SchedulerMenuItemId.NewAppointment)
                If item IsNot Nothing Then
                    item.Caption = "&New Meeting"
                    item.Image = Image.FromFile("NewItem_16x16.png")
                End If

        ' Create a menu item for the Scheduler command.
        Dim service As ISchedulerCommandFactoryService = schedulerControl1.GetService(Of ISchedulerCommandFactoryService)()
        Dim cmd As SchedulerCommand = service.CreateCommand(SchedulerCommandId.SwitchToGroupByResource)
        Dim menuItemCommandAdapter As New SchedulerMenuItemCommandWinAdapter(cmd)
        Dim menuItem As DXMenuItem = CType(menuItemCommandAdapter.CreateMenuItem(DXMenuItemPriority.Normal), DXMenuItem)
        menuItem.BeginGroup = True
        e.Menu.Items.Add(menuItem)

        ' Insert a new item into the Scheduler popup menu and handle its click event.
        e.Menu.Items.Add(New SchedulerMenuItem("Click me!", AddressOf MyClickHandler))
    End If
End Sub

Public Sub MyClickHandler(ByVal sender As Object, ByVal e As EventArgs)
    MessageBox.Show("My Menu Item Clicked!")
End Sub
#End Region ' #popupmenushowing
    End Class
End Namespace